pylot/planning/frenet_optimal_trajectory/FrenetOptimalTrajectory/src/CubicSpline/CubicSpline1D.cpp

```diff
@@ -0,0 +1,107 @@
+#include <algorithm>
+#include <numeric>
+#include <cmath>
+#include "CubicSpline1D.h"
+
+using namespace std;
+using namespace Eigen;
+
+// Default constructor
+CubicSpline1D::CubicSpline1D() = default;
+
+// Construct the 1-dimensional cubic spline.
+CubicSpline1D::CubicSpline1D(vector<double>& v1, vector<double>& v2) {
```

Comment:
One syntax to make more explicit the initialization of attributes is:
`CubicSplie1D(vector<double>& v1, vector<double>& v2) : x (v1), y(v2), nx(v1.size()) {
}`

Reply:
Done, added to other files where appropriate.